src/main/java/com/google/devtools/build/lib/remote/disk/DiskCacheGarbageCollectorIdleTask.java

```diff
@@ -93,10 +93,10 @@ public void run() {
       logger.atInfo().log("Disk cache garbage collection started");
       CollectionStats stats = gc.run();
       logger.atInfo().log("%s", stats.displayString());
-    } catch (IOException e) {
-      logger.atInfo().withCause(e).log("Disk cache garbage collection failed");
     } catch (InterruptedException e) {
       logger.atInfo().withCause(e).log("Disk cache garbage collection interrupted");
+    } catch (Throwable e) {
```

Comment:
`Throwable` should not be caught as `Error`s aren't guaranteed to be recoverable. If this change avoids the OOME, wouldn't that be sufficient?

Reply:
+1, I don't think we should catch `Throwable` here.

Reply:
The catch was not for recovering, but for easier diagnostics. Without it will not show the second line in:
```
241027 14:46:16.906:I 480 [com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask.run] Disk cache garbage collection started
241027 14:47:12.220:WT 480 [com.google.devtools.build.lib.remote.disk.DiskCacheGarbageCollectorIdleTask.run] Disk cache garbage collection failed
```
if there is some other unexpected error in the future.